### DISCUSSION TITLE
request object from Flask is not populated when calling a validation function on connexion 3.x

### DISCUSSION THREAD

**Discussion by nodirsid** [spec-first/connexion#1816] [General]
When running a Flask based webapp with `connexion 2.x`, the `request` object is usually populated upon triggering the validation function. I define this validation function in API security definition in yml file as `x-basicInfoFunc` as part of authentication flow. Using Python 3.11.

Now, when upgrading to `connexion 3.x` the same Flask based `request` object is not populated. When trying to access this object I get the following `RuntimeError`:

`RuntimeError: Working outside of operation context. This typically means that you attempted to use functionality that needed an active HTTP request. Consult the documentation on testing for information about how to avoid this problem.`

The expectation is Flask's `request` should populate when using the same authentication flow in `connexion 3.x` and not throw above error.

Has anyone faced the same issue yet?

**Comment by RobbeSneyders** [comment]
Hi @nodirsid, thanks for the report. 

This would be fixed by #1750, which would expose the Connexion Request to the security validation functions. I'll try to prioritize it, but if it's blocking, you can look into opening a PR as well.

**Reply by nodirsid** [reply]
Thank you @RobbeSneyders for your swift reply and confirmation of the issue. Do you have any timelines when #1750 can be generally available?

Also, thank you for offering to raise a PR to contribute to an open source community.

**Comment by chrisinmtown** [comment]
Very late comment. I also noticed that this behavior changed v2 -> v3. The latest version of Connexion v3 injects the `request` object into a securityScheme handling function if that function declares a parameter with that name. This works for me.

So if you don't mind a suggestion, maybe it's time to close this ticket?
